src/frontend/instruction_analyzer.rs

```diff
@@ -0,0 +1,193 @@
+    fn has_memory_read(&self) -> bool {
+        self.info().expect("Unable to get InstructionInfo").reads.iter().any(|x| match x { 
+            &IOperand::Memory { base: _, index: _, scale: _, disp: _ } => true,
+            _ => false,
+        })
+    }
+
+    /// Return if this instruction writes to memory
+    fn has_memory_written(&self) -> bool {
+        self.info().expect("Unable to get InstructionInfo").writes.iter().any(|x| match x { 
+            &IOperand::Memory { base: _, index: _, scale: _, disp: _ } => true,
+            _ => false,
+        })
+    }
+
+    /// Returns memory written to by the instruction, in standard sib form
+    fn memory_written(&self) -> Option<&IOperand> {
+        self.info().expect("Unable to get InstructionInfo").writes.iter().find(|&x| match x {
+            &IOperand::Memory { base: _, index: _, scale: _, disp: _ } => true,
+            _ => false,
+        })
+    }
+
+    /// Returns memory read by the instruction, in standard sib form
+    fn memory_read(&self) -> Option<&IOperand> {
+        self.info().expect("Unable to get InstructionInfo").reads.iter().find(|&x| match x {
+            &IOperand::Memory { base: _, index: _, scale: _, disp: _ } => true,
+            _ => false,
+        })
+    }
+}
+
+/// Capstone-based x86 instruction analyzer
+pub struct X86_CS_IA {
+    bytes: Vec<u8>,
+    cs: cs::Capstone,
+    info: InstructionInfo,
+}
+
+//impl From<cs::CsErr> for IAError {
+    //fn from(err: cs::CsErr) -> IAError {
+        //Cow::from(err.to_string())
+    //}
+//}
+
+impl X86_CS_IA {
+
+    fn reg_name(&self, reg: u32) -> Option<String> {
+        self.cs.reg_name(reg).map(|x| x.to_owned())
+    }
+
+    fn analyze(&mut self) -> Result<(), IAError> {
+        // Disassembly exactly one instruction
+        let buf = match self.cs.disasm(self.bytes.as_slice(), 0x0, 1) {
+            Ok(d) => d,
+            Err(e) => return Err(Cow::from(e.to_string())),
+        };
+
+        for instr in buf.iter() {
+            // TODO: Look into prefixes for opcode.
+            // See how this information is set in capstone.
+            self.info.mnemonic = Cow::from(instr.mnemonic);
+            let details = instr.detail.unwrap();
+            if let cs::DetailsArch::X86(arch) = details.arch {
+                for i in 0..arch.op_count {
+                    // Get the current operand
+                    let op: cs::cs_x86_op = arch.operands[i as usize];
+                    let iop: IOperand = match op.type_ {
+                        cs::x86_op_type::X86_OP_REG => {
+                            let reg: cs::x86_reg = op.reg();
+                            if let Some(reg_name) = self.reg_name(reg.as_int()) {
+                                IOperand::Register(reg_name)
+                            } else {
+                                IOperand::Invalid
+                            }
+                        },
+                        cs::x86_op_type::X86_OP_MEM => {
```

Comment:
This information is available in `ao[j]` output. If you need detailed information - there is an 'opex' field.